src/lib/services/segment-schema.ts

```diff
@@ -0,0 +1,19 @@
+import joi from 'joi';
+import { constraintSchema } from '../schema/feature-schema';
+
+export const segmentSchema = joi
+    .object()
+    .keys({
+        name: joi.string().required(),
+        description: joi.string().allow(null).allow('').optional(),
```

Comment:
Could this be an issue in more strictly typed languages? If something can be a null or a string?

Reply:
Looks like we usually do `.allow(null).allow('').optional()` for optional string fields.

Reply:
True. And it makes sense if we never expose this in the client API. But if we for some reason want to expose fields like this in the client API in the future we might have a problem. Is there any reason we can't default it to empty string instead?